src/compositions/question-form-4.tsx

```diff
@@ -0,0 +1,100 @@
+import { FC, useState } from 'react';
+import { InputField } from '../elements/input-field';
+import { useRouter } from 'next/dist/client/router';
+import { RadioGroup } from '@headlessui/react';
+
+type Props = {
+  label: string;
+  initialAnswer?: string;
+  href: string;
+  answer: { [key: string]: string };
+  callback: React.Dispatch<React.SetStateAction<{ [key: string]: string }>>;
+};
+const validateInput = (answer: string) => {
+  if (parseFloat(answer) > 10000) {
+    return 'Bist dir sicher? Über 10000 m2 für weniger als 251 Mitarbeiter scheint mir nach zu viel Büro.';
+  } else if (answer.length > 10) {
+    return 'Bitte gib maximal 10 Ziffern nach dem Komma ein.';
+  }
+  return null;
+};
+
+export const Question4: FC<Props> = ({ label, href, answer, callback }) => {
+  const router = useRouter();
+  const answerQuestion4 = answer?.kWh === undefined ? { kWh: '', electricityType: '' } : answer;
+
+  return (
+    <div className="mb-8">
+      <InputField
+        type="number"
+        label={label}
+        name="answer"
+        id="number"
+        step="1"
+        placeholder="5000"
+        min="0"
+        max="250"
+        value={answerQuestion4?.kWh || ''}
+        onChange={(value) => {
+          callback({ ...answerQuestion4, kWh: value });
+        }}
+        onKeyDown={(key) => key === 'Enter' && router.push(href)}
+        validateInput={validateInput}
+      />
+      <RadioButton callback={callback} answerQuestion4={answerQuestion4} />
+    </div>
+  );
+};
+
+const electricityOptions = ['Ökostrom', 'Nöd Öko', 'Weiss nöd'];
```

Comment:
You might probably want to separate your labels from your values. `Nöd Öko` works as a text but not really as a value.